src/plugins/auto/plugin.cpp

```diff
@@ -719,6 +719,14 @@ std::vector<DeviceInformation> MultiDeviceInferencePlugin::FilterDevice(const st
 }
 std::vector<DeviceInformation> MultiDeviceInferencePlugin::FilterDeviceByNetwork(const std::vector<DeviceInformation>& metaDevices,
                                                 InferenceEngine::CNNNetwork network) {
+    if (metaDevices.empty()) {
+        IE_THROW(NotFound) << "No available device to filter " << GetName() <<  " plugin";
+    }
+
+    if (metaDevices.size() == 1) {
+        return metaDevices;
+    }
```

Comment:
why do we need it? with a lot of `if else` the code become less readable. Better to have generic logic..

Reply:
@ilya-lavrenov In 2022.1, AUTO only support dynamic network on CPU for stabilization. 
In 2022.2, AUTO will support dynamic network on GPU when GPU is only candidate device. We would like to enable this feature first in 2022.2 with less change as w/a by this PR.  
We will do reconstruction later to refine all generic logic in AUTO.

Reply:
shall we enable dynamic shape test path for this case?